docs/nodejs_dev_guide.rst

```diff
@@ -0,0 +1,205 @@
+Node.js Development Guide
+=========================
+
+By default, Node.js is already installed on the `teracy-dev` VM with the following configuration:
+
+..  code-block:: json
+
+    {
+      "nodejs":{
+        "enabled":true,
+        "version":"0.10.28",
+        "checksum":"abddc6441e0f208f6ed8a045e0293f713ea7f6dfb2d6a9a2024bf8b1b4617710",
+        "npm": {
+          "version":"1.4.3",
+          "globals": [
+            {
+              "name":"grunt-cli",
+              "version":""
+            },
+            {
+              "name":"gulp",
+              "version":""
+            },
+            {
+              "name":"bower",
+              "version":""
+            },
+            {
+              "name":"yo",
+              "version":""
+            },
+            {
+             "name":"phantomjs",
+             "version":""
+            }
+          ]
+        }
+      }
+    }
+
+Make sure you have the teracy-dev VM running (``$ vagrant up``) by following the
+:doc:`getting_started` guide.
+
+
+Verify Node.js
+--------------
+
+``$ vagrant ssh`` and check:
+
+#.  ``node``
+
+    ..  code-block:: bash
+
+        $ node --version
+        v0.10.28
+
+#.  ``npm``
+
+    ..  code-block:: bash
+
+        $ npm --version
+        1.4.3
+
+
+Hello World
+-----------
+
+We're going to follow the example from https://nodejs.org/ to create a `Hello World` web server.
+
+..  code-block:: bash
+
+    $ ws
+    $ cd personal
+    $ mkdir node-hello
+    $ cd node-hello
+
+Let's use `vim` to create a file named `example.js` under the `node-hello` directory with the following
+content:
+
+..  code-block:: javascript
+
+    var http = require('http');
+    http.createServer(function (req, res) {
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      res.end('Hello World\n');
+    }).listen(3000, '0.0.0.0');
+    console.log('Server running at http://0.0.0.0:3000/');
+
+Run it with:
+
+..  code-block:: bash
+
+    $ node example.js
+    Server running at http://0.0.0.0:3000/
+
+And then open the browser at http://localhost:3000 and we should see the `Hello World` displayed.
+
+We can start digging `Node.js` now at: https://nodejs.org/documentation/
+
+
+Express Framework
+-----------------
+
+We're going to follow `Getting started` section from http://expressjs.com/starter/hello-world.html
+to create a `Hello World` web application.
+
+
+..  code-block:: bash
+
+    $ ws
+    $ cd personal
+    $ mkdir myapp
+    $ cd myapp
+    $ npm init
+    $ npm install express --save
```

Comment:
I got this error message when run the above command: "Please try running this command again as root/Administrator".
This error might not always happen but we should change it to `sudo npm install express --save` to avoid it.